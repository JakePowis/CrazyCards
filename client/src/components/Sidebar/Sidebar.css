.sidebar{
    position: fixed;
    transition: all 0.5s;
    transform: translateX(-35rem);
    width: 35rem;
    background-color: #f0f0f0;
    height: 100%;
  }
  
  .sidebar-cats {
    /* padding-left: 2rem; */
  
  }
   .sidebar-cats > h3 {
    margin-top: 4rem;
    margin-bottom: 0rem;
    padding-left: 2rem;
  }
  
  
  .sidebar > .welcome > a {
  
    color: white;
    font-size: 3rem;
    text-align: center;
    padding: 1rem;
    margin-right:1.5rem;
    margin-top:2.5rem;
  }
  
  .sidebar > .welcome > i{
  
    margin-top:1.8rem;
  }
  
  .sidebar.open{
    transform: translateX(0);
  }
  
  .sidebar-close-button{
    border-radius: 50%;
    border: .1rem #000000 solid;
    width: 3rem;
    height: 3rem;
    padding: .5rem;
    font-size: 2rem;
    padding-top: 0;
    cursor: pointer;
    position: absolute;
    right: .5rem;
    top: 1.5rem
  }