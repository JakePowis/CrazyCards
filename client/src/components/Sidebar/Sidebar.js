import React from 'react'
import { Link } from 'react-router-dom';
import './Sidebar.css';

export default function Sidebar() {

    const closeMenu = () => {
        document.querySelector(".sidebar").classList.remove("open")
      }
    

    return (
        <aside className="sidebar">


        <div className="sidebar-cats">
          <h3>Our Cards</h3>
          <button className="sidebar-close-button" onClick={closeMenu}>x</button>
          <ul className="categories">

          <li>  <Link to="/card/student">Student Life</Link></li>
          <li><Link to="/card/anywhere">Anywhere Card</Link></li>
          <li><Link to="/card/liquid">Liquid Card</Link></li>
            {/* <li>
              <Link to="/shop">All</Link>
            </li>
            <li>
              <Link to="/category/fantasy">Fantasy</Link>
            </li>

            <li>
              <Link to="/category/crime">Crime</Link>
            </li>
            <li>
              <Link to="/category/romance">Romance</Link>
            </li>
            <li>
              <Link to="/category/nonfiction">Non-fiction</Link>
            </li>
          </ul>

          <h3>Sale</h3>
          <ul className="categories">
            <li>
              <Link to="/sale">Sale Items</Link>
            </li>
          </ul>


          <h3>Account</h3>

          <ul className="categories">
            <li>
              <Link to="/customerservice">Customer Services</Link>
            </li>

            <li>
              <Link to="/profile">Profile</Link>
            </li>
            <li>
              <Link to="/profile">Sign Out</Link>
            </li> */}
          </ul>
        </div>



      </aside>
    )
}
