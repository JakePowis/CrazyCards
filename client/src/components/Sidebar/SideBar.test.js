import React from 'react';
import ReactDom from 'react-dom';
import Sidebar from './Sidebar.js'
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from "react-test-renderer"

afterEach(cleanup)

test("Sidebar Renders Correctly", ()=> {
    const div = document.createElement("div");
    ReactDom.render( <Router> <Sidebar /></Router>, div)
})

test('Renders close button', () => {
    const { getByTestId } = render(
        <Router>
            <Sidebar />
        </Router>);
    const divElement = getByTestId("close");
    expect(divElement).toBeInTheDocument();
  });

  test("matches snapshot", ()=> {
    const tree = renderer.create(<Router> <Sidebar /></Router>).toJSON()
    expect(tree).toMatchSnapshot();
})
  