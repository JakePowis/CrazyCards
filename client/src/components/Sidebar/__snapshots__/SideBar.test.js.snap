// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  " ",
  <aside
    className="sidebar"
  >
    <button
      className="sidebar-close-button"
      data-testid="close"
      onClick={[Function]}
    >
      x
    </button>
    <h3>
      Home
    </h3>
    <ul
      className="cards"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <li
          onClick={[Function]}
        >
          <span
            className="indent"
          >
            Match a Card
          </span>
        </li>
      </a>
    </ul>
    <h3>
      Our Cards
    </h3>
    <ul
      className="cards"
    >
      <a
        href="/card/1"
        onClick={[Function]}
      >
        <li>
          <span
            className="indent"
          >
            Student Life
          </span>
        </li>
      </a>
      <a
        href="/card/2"
        onClick={[Function]}
      >
          
        <li>
          <span
            className="indent"
          >
            Anywhere Card
          </span>
        </li>
      </a>
      <a
        href="/card/3"
        onClick={[Function]}
      >
         
        <li>
          <span
            className="indent"
          >
            Liquid Card
          </span>
        </li>
      </a>
    </ul>
  </aside>,
]
`;
